puro/lib/src/version.dart

```diff
@@ -80,8 +81,8 @@ class PuroVersion {
       final packages = packageData['packages'] as List<dynamic>;
       final puroPackage = packages
           .cast<Map<String, dynamic>>()
-          .firstWhere((e) => e['name'] == 'puro');
-      final rootUri = Uri.parse(puroPackage['rootUri'] as String);
+          .firstWhereOrNull((e) => e['name'] == 'puro');
+      final rootUri = Uri.parse(puroPackage?['rootUri'] as String);
```

Comment:
the `as String` will fail if puroPackage is null, you should just check and return null here

Reply:
Fixed Now.

Reply:
the `&& puroPackage?['rootUri'] == null` part is unnecessary, it will always be null if puroPackage is null

Reply:
updated code as required.